tests/test_model.py

```diff
@@ -102,3 +102,22 @@ def test_predict_success_with_file(*args):
 
     assert np.array_equal(result, expected)
     assert req_id == "123"
+
+
+@patch("giza_actions.model.GizaModel._set_session")
+@patch("giza_actions.model.GizaModel._get_output_dtype")
+@patch("giza_actions.model.GizaModel._retrieve_uri")
+@patch("giza_actions.model.GizaModel._get_endpoint_id", return_value=1)
+def test_cache_implementation(*args):
```

Comment:
The test is failing as some mocks are missing, the current error is due to missing credentials, for that just add:

```python
@patch("giza_actions.model.GizaModel._get_credentials")
def test_cache_implementation(*args):
```

Also it is possible that some patches are missing as well, for example for `_get_model` and `_get_version`  might be need.